### PR TITLE
[DSVA-15744] - ASPP and CBAM (attention) improvement

### PR DESCRIPTION
**Pull request by Pawel842**

### CODE REVIEW

File: yolox/models/network_blocks.py [hunk 1]
```diff
@@ -144,6 +144,78 @@ def forward(self, x):
...
+        global_features = self.global_branch(x)
+        global_features = nn.functional.interpolate(
+            global_features, size=x.shape[2:], mode='bilinear', align_corners=False
+        )
+
+        # Concatenate all features
+        x = torch.cat([x1] + atrous_branches + [global_features], dim=1)
+
+        # Apply dropout
+        x = self.dropout(x)
```
Comment (beazy96) [inline yolox/models/network_blocks.py:213]: 🟡Why dropout here? Its not mentioned in paper

File: yolox/models/network_blocks.py [hunk 2]
```diff
@@ -144,6 +144,78 @@ def forward(self, x):
...
 
 
+class ASPPBottleneck(nn.Module):
+    """Atrous Spatial Pyramid Pooling layer"""
+
+    def __init__(
+            self, in_channels, out_channels, dilation_rates=(6, 12, 18), activation="silu", dropout_rate=0.1
+    ):
+        super().__init__()
+        hidden_channels = in_channels // 2
```
Comment (beazy96) [inline yolox/models/network_blocks.py:154]: 🟡 Its not mentioned in paper. Have you tested without it?

File: yolox/models/network_blocks.py [hunk 3]
```diff
@@ -144,6 +144,78 @@ def forward(self, x):
...
+        self.conv1 = BaseConv(in_channels, hidden_channels, 1, stride=1, act=activation)
+
+        # Regular 1x1 convolution branch
+        self.branch1 = BaseConv(hidden_channels, hidden_channels, 1, stride=1, act=activation)
+
+        # Atrous convolution branches
+        self.branches = nn.ModuleList()
+        for rate in dilation_rates:
```
Comment (beazy96) [inline yolox/models/network_blocks.py:162]: 🟡In paper mentioned only `rate=3`. Have you tested it?

**Review by beazy96 — CHANGES_REQUESTED** [review]

File: yolox/models/network_blocks.py (hunk 1 above)
Comment (Pawel842) [reply yolox/models/network_blocks.py:213]: I added dropout to improve the results, there were a little better with it

File: yolox/models/network_blocks.py (hunk 3 above)
Comment (Pawel842) [reply yolox/models/network_blocks.py:162]: no, I couldn't find this info, but I see it now, I will run in ;)

File: yolox/models/network_blocks.py (hunk 2 above)
Comment (Pawel842) [reply yolox/models/network_blocks.py]: hmm, It was in the implementation I found but I will remove it and run

File: yolox/models/network_blocks.py (hunk 3 above)
Comment (Pawel842) [reply yolox/models/network_blocks.py:167]: The best results (mAP 0:78) were from the implementations with current combination of rates

File: yolox/models/network_blocks.py (hunk 2 above)
Comment (Pawel842) [reply yolox/models/network_blocks.py:159]: The best results (mAP 0:78) were from the implementations with current number of channels

**Review by beazy96 — APPROVED** [review]

**Comment by Pawel842** [comment]
@pkolebski  up:)

**Review by pkolebski — APPROVED** [review]
